libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java

```diff
@@ -277,6 +277,7 @@ public void addProgressChangeListener(MapboxNavigation navigation) {
   public void addProgressChangeListener(MapboxNavigation navigation, boolean vanishRouteLineEnabled) {
     this.navigation = navigation;
     this.vanishRouteLineEnabled = vanishRouteLineEnabled;
+    this.mapRouteProgressChangeListener = buildMapRouteProgressChangeListener();
```

Comment:
🤔  this is also done in [a constructor](https://github.com/mapbox/mapbox-navigation-android/blob/73e3421d62e82d9ec2b8fb5655b40529436bc582/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java#L102) 

Also confusing why we want to add it to this `addProgressChangeListener`, but not to the other [addProgressChangeListener](https://github.com/mapbox/mapbox-navigation-android/blob/73e3421d62e82d9ec2b8fb5655b40529436bc582/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java#L262-L265)

Reply:
are these functions supposed to call [updateProgressChangeListener](https://github.com/mapbox/mapbox-navigation-android/blob/73e3421d62e82d9ec2b8fb5655b40529436bc582/libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java#L410-L418)?

Reply:
The vanishing route line is off by default so the other call to addProgressChangeListener doesn't change anything.

This fix is kind of a retrofit for some other changes that recently. I didn't want to refactor the entire NavigationMapboxMap class in order use the builder that was added to NavigationMapRoute.